test/insert.bats

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bats -t
+# umoci: Umoci Modifies Open Containers' Images
+# Copyright (C) 2018 Cisco
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+load helpers
+
+function setup() {
+	setup_image
+}
+
+function teardown() {
+	teardown_tmpdirs
+	teardown_image
+}
+
+@test "umoci insert" {
+	image-verify "${IMAGE}"
+
+	# fail with too few arguments
+	umoci insert --image "${IMAGE}:${TAG}"
+	[ "$status" -ne 0 ]
+
+	# ...and too many
+	umoci insert --image "${IMAGE}:${TAG}" asdf 123 456
+	[ "$status" -ne 0 ]
+
+	# do an insert
+	umoci insert --image "${IMAGE}:${TAG}" "${ROOT}/test/insert.bats" /tester
+	[ "$status" -eq 0 ]
+
+	umoci insert --image "${IMAGE}:${TAG}" "${ROOT}/umoci" /tester
+	[ "$status" -eq 0 ]
```

Comment:
I would also recommend having a rootfs with something like

```
mkdir -p $BUNDLE_A/rootfs/some/path
chmod 000 $BUNDLE_A/rootfs/some/path
umoci repack # whatever
umoci insert blah /some/path/blah
```

(This should make no difference to the actual generation, but it would ensure that rootless is "handled" in that case. Also a similar case should be handled from the host side of things to make sure rootless is properly obeyed and the added directory is actually reconstructed correctly.)

Also this test should do an `unpack` after the first insertion to ensure that the file actually changes between the two insertions IMO.

If all of that is a bit too much to change, I can push them on top of your branch once you're done developing.

Reply:
Nope, I'm happy to do that too.